synthesizer/process/src/lib.rs

```diff
@@ -66,30 +66,61 @@ use synthesizer_program::{
 };
 use synthesizer_snark::{ProvingKey, UniversalSRS, VerifyingKey};
 
-use aleo_std::prelude::{finish, lap, timer};
-use indexmap::IndexMap;
-use parking_lot::RwLock;
-use std::{collections::HashMap, sync::Arc};
+use aleo_std::{
+    StorageMode,
+    prelude::{finish, lap, timer},
+};
+use lru::LruCache;
+use parking_lot::{Mutex, RwLock};
+use std::{collections::HashMap, num::NonZeroUsize, sync::Arc};
+use tracing::debug;
 
 #[cfg(feature = "aleo-cli")]
 use colored::Colorize;
 
+#[cfg(not(feature = "rocks"))]
+use ledger_store::helpers::memory::ConsensusMemory;
+#[cfg(feature = "rocks")]
+use ledger_store::helpers::rocksdb::ConsensusDB;
+
+#[cfg(feature = "rocks")]
+#[derive(Clone)]
+pub struct Process<N: Network> {
+    /// The universal SRS.
+    universal_srs: Arc<UniversalSRS<N>>,
+    /// The Stack for credits.aleo
+    credits: Option<Arc<Stack<N>>>,
+    /// The mapping of program IDs to stacks.
+    stacks: Arc<Mutex<LruCache<ProgramID<N>, Arc<Stack<N>>>>>,
+    /// The storage.
+    store: Option<ConsensusStore<N, ConsensusDB<N>>>,
+}
+
+#[cfg(not(feature = "rocks"))]
 #[derive(Clone)]
 pub struct Process<N: Network> {
     /// The universal SRS.
     universal_srs: Arc<UniversalSRS<N>>,
+    /// The Stack for credits.aleo
+    credits: Option<Arc<Stack<N>>>,
     /// The mapping of program IDs to stacks.
-    stacks: IndexMap<ProgramID<N>, Arc<Stack<N>>>,
+    stacks: Arc<Mutex<LruCache<ProgramID<N>, Arc<Stack<N>>>>>,
+    /// The storage.
+    store: Option<ConsensusStore<N, ConsensusMemory<N>>>,
 }
 
 impl<N: Network> Process<N> {
     /// Initializes a new process.
     #[inline]
     pub fn setup<A: circuit::Aleo<Network = N>, R: Rng + CryptoRng>(rng: &mut R) -> Result<Self> {
         let timer = timer!("Process:setup");
-
         // Initialize the process.
-        let mut process = Self { universal_srs: Arc::new(UniversalSRS::load()?), stacks: IndexMap::new() };
+        let mut process = Self {
+            universal_srs: Arc::new(UniversalSRS::load()?),
+            credits: None,
+            stacks: Arc::new(Mutex::new(LruCache::new(NonZeroUsize::new(N::MAX_STACKS).unwrap()))),
+            store: None,
```

Comment:
since the store is now a possible member, perhaps it would be a good idea to reflect it in this function name that it doesn't produce one? e.g. `setup_wo_store` or `new_without_store`